githubutils/commitutils/branch.go

```diff
@@ -0,0 +1,111 @@
+package commitutils
+
+import (
+	"context"
+	"github.com/google/go-github/github"
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/go-utils/vfsutils"
+	"go.uber.org/zap"
+	"time"
+)
+
+var (
+	RefNotSetError = errors.Errorf("Must initialize with set ref before updating files.")
+	RefAlreadySetError = errors.Errorf("Ref was already set.")
+)
+
+type CommitSpec struct {
+	Name    string
+	Email   string
+	Message string
+}
+
+type RefUpdater interface {
+	SetRef(ctx context.Context, ref *github.Reference) error
+	UpdateFile(ctx context.Context, path string, contentUpdater func(string) string) error
+	Commit(ctx context.Context, spec CommitSpec) error
+}
+
+type githubRefUpdater struct {
+	client *github.Client
```

Comment:
This code assumes a client exists from the beginning, what if the installation_id is passed through the instead?